### PR TITLE
Fix/stealfocus

### LINKED ISSUE #2123: [new GUI] steal focus by the webview
https://github.com/readium/r2-navigator-js/commit/f111fa2109c36f616a30c7099fe6d57f99b260ab#diff-2a13f350e34666a1b6a335367fc2ad1c3304923ba63879e1cf043e7d5bcd66deR251

### PR DESCRIPTION
**Pull request by panaC**
fixes #2123

### CODE REVIEW

**Comment by panaC** [comment]
complete and finish PR https://github.com/edrlab/thorium-reader/pull/2117

**Review by danielweck — CHANGES_REQUESTED** [review]

File: src/renderer/reader/components/ReaderMenu.tsx [hunk 1]
```diff
@@ -1236,62 +1204,28 @@ export const ReaderMenu: React.FC<IBaseProps> = (props) => {
...
-                    // tabModeRef.current.removeEventListener('focusout',  tabmodeFocusOUT);
-                    // tabModeRef.current.addEventListener('focusout', tabmodeFocusOUT);
-
-                    console.log("Focus on tabmode");
-                    tabModeRef.current.focus();
-                } else {
-                    console.error("!no tabModeRef on tabmode");
-                }
-            }
+            }, 1000);
```
[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderMenu.tsx]: e29520dcceb98752bc3c556787cc07aec562dee9

File: src/renderer/reader/components/ReaderHeader.tsx [hunk 2]
```diff
@@ -953,7 +953,19 @@ export class ReaderHeader extends React.Component<IProps, IState> {
...
+                                    console.log("MENU DialogOnOpenChange", open);
+                                    this.props.handleMenuClick(open);
+                                    if (open) {
+                                        stealFocusDisable(true);
+                                    } else {
+                                        stealFocusDisable(false);
+                                    }
+                                }}
+                                modal={this.props.readerPopoverDialogContext.dockedMode ? false : true}
+                            >
```
[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderHeader.tsx:968]: e29520dcceb98752bc3c556787cc07aec562dee9

File: src/renderer/reader/components/ReaderHeader.tsx [hunk 3]
```diff
@@ -953,7 +953,19 @@ export class ReaderHeader extends React.Component<IProps, IState> {
 
                             {/* { this.props.menuOpen ? */}
 
-                            <Dialog.Root open={this.props.menuOpen} onOpenChange={(v) => { console.log("MENU DialogOnOpenChange", v); this.props.handleMenuClick(v); }} modal={this.props.readerPopoverDialogContext.dockedMode ? false : true}>
+                            <Dialog.Root
+                                open={this.props.menuOpen}
+                                onOpenChange={(open) => {
+                                    console.log("MENU DialogOnOpenChange", open);
+                                    this.props.handleMenuClick(open);
+                                    if (open) {
```
[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: Partially fixed with this commit https://github.com/edrlab/thorium-reader/pull/2143/commits/f9c0ddd861a55d57d8c0affaa15a2474d23d365f

[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: On not docked mode (fully dialog modal) : SHIFT + ALT + CLIC on an hyperlink doesn't close the modal. You must then click on the cross or hit escape key to close the modal, the focus isn't prevented to move to the trigger

[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: we can do the reverse in docked mode : aka SHIFT + ALT + CLIC on an hyperlink close the docked dialog and focus back to the main area @danielweck

[the remark this answers is not in the document]
Comment (panaC) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: https://github.com/edrlab/thorium-reader/blob/f9c0ddd861a55d57d8c0affaa15a2474d23d365f/src/renderer/reader/components/ReaderMenu.tsx#L146-L150

[the remark this answers is not in the document]
Comment (danielweck) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: Confirmed it was SHIFT + ALT in 2.4

https://github.com/edrlab/thorium-reader/blob/v2.4.1/src/renderer/reader/components/ReaderMenu.tsx#L251-L267

[the remark this answers is not in the document]
Comment (danielweck) [reply src/renderer/reader/components/ReaderHeader.tsx:964]: Note: also for ENTER key activation of TOC hyperlink (or bookmark, etc.) not just mouse click

**Comment by danielweck** [comment]
Hello @panaC I propose that we merge this PR and later iterate on the subtle keyboard-driven navigation (SHIFT ALT CLICK on link or with ENTER key, for example). This PR branch has a code conflict, could you resolve it please? (I am not sure how to myself, you know the impacted code chunk better since you implemented the mods in the first place ... I don't want to break something :)

**Comment by panaC** [comment]
> Hello @panaC I propose that we merge this PR and later iterate on the subtle keyboard-driven navigation (SHIFT ALT CLICK on link or with ENTER key, for example). This PR branch has a code conflict, could you resolve it please? (I am not sure how to myself, you know the impacted code chunk better since you implemented the mods in the first place ... I don't want to break something :)

Great, This is what i was going to do !

**Pull request merged**